### PR TITLE
fix(ivy): errors not being logged to ErrorHandler

### PR DESCRIPTION
**Pull request by crisbeto**
Fixes Ivy not passing thrown errors along to the `ErrorHandler`.

**Note:** the failing test had to be reworked a little bit, because it has some assertions that depend on an error context being logged, however Ivy doesn't keep track of the error context.

This PR resolves FW-840.

### CODE REVIEW

**Review by kara — CHANGES_REQUESTED** [review]

File: packages/core/src/render3/instructions.ts [hunk 1]
```diff
@@ -931,14 +932,24 @@ export function componentHostSyntheticListener<T>(
...
-    eventName: string, listenerFn: (e?: any) => any, useCapture = false,
+    eventName: string, callback: (e?: any) => any, useCapture = false,
     eventTargetResolver?: GlobalTargetResolver,
     loadRendererFn?: ((tNode: TNode, lView: LView) => Renderer3) | null): void {
   const lView = getLView();
   const tNode = getPreviousOrParentTNode();
   const tView = lView[TVIEW];
   const firstTemplatePass = tView.firstTemplatePass;
   const tCleanup: false|any[] = firstTemplatePass && (tView.cleanup || (tView.cleanup = []));
+  const listenerFn = function(event?: any) {
```
[the remark this answers is not in the document]
Comment (mhevery) [reply packages/core/src/render3/instructions.ts]: `wrapListenerWithPreventDefault` is probably best bet.

**Review by mhevery — CHANGES_REQUESTED** [review]

**Comment by crisbeto** [comment]
Feedback has been addressed.

**Comment by kara** [comment]
@crisbeto As discussed, needs rebase and logic moved into existing wrapper in `wrapListener`

**Review by kara — APPROVED** [review]
LGTM, once CI issues are resolved (they look unrelated)

**Review by mhevery — APPROVED** [review]

**Comment by kara** [comment]
[presubmit](http://test/OCL:232301472:BASE:232301501:1549297663599:a305b402)

**Pull request closed without merging**